### PR TITLE
feat: allow default node pools metadata key-value pairs to be disabled

### LINKED ISSUE #2004: Importing node pools can forces replacement
### TL;DR

The module is adding two key-value pairs `cluster_name` and `node_pool` that triggers a replacement if the node pool was not provisioned using the community module.

GKE [node_pools_metadata](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/autogen/main/cluster.tf.tmpl#L894)

https://github.com/hashicorp/terraform-provider-google/issues/1388

### Expected behavior

Allow users to fully overwrite node pools metadata or expose a way to disable key-pair values added by default

### Observed behavior

The node pool is marked to be replaced
```terraform
            ...
            local_ssd_count             = 0
            logging_variant             = "DEFAULT"
            machine_type                = "t2d-standard-16"
          ~ metadata                    = { # forces replacement
              + "cluster_name"             = "admin"
                "disable-legacy-endpoints" = "true"
              + "node_pool"                = "runners"
            ...
            }
```

### Terraform Configuration

```hcl
import {
  id = "projects/project-id/locations/us-east1/clusters/admin"
  to = module.cluster.module.gke.google_container_cluster.primary
}

import {
  id = "project-id/us-east1/admin/runners"
  to = module.cluster.module.gke.google_container_node_pool.pools["runners"]
}

module "cluster" {
  source                     = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  version                    = "30.0.0"
  ip_range_pods              = var.ip_range_pods
  ip_range_services          = var.ip_range_services
  master_ipv4_cidr_block     = var.master_ipv4_cidr_block
  name                       = "admin"
  network                    = "default"
  project_id                 = "project-id"
  region                     = "us-east1"
  subnetwork                 = "default"
  deletion_protection        = true
  master_authorized_networks = var.master_authorized_networks

  node_pools = [
    {
      name               = "runners"
      machine_type       = "t2d-standard-16"
      initial_node_count = 0
      min_count          = 0
      max_count          = 20
      enable_secure_boot = true
      spot               = true
      location_policy    = "ANY"
      effect             = "NO_SCHEDULE"
      key                = "node_type"
      value              = "runners"
    }
  ]
}
```

### Terraform Version

```sh
Terraform v1.7.4 on darwin_arm64
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by caueasantos**
Resolves #2004 

There are many ways to go about this, but I tried to quickly expose a way for users to disable default metadata labels added by the module.

First time opening a pull request here, let me know if I missed anything.

### CODE REVIEW

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @caueasantos!

Should this change apply to Autopilot cluster?

```
Working in ./modules/beta-autopilot-private-cluster ...
1 issue(s) found:

Warning: [Fixable] variable "enable_default_node_pools_metadata" is declared but not used (terraform_unused_declarations)

  on variables.tf line 427:
 427: variable "enable_default_node_pools_metadata" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.7.0/docs/rules/terraform_unused_declarations.md

tflint failed ./modules/beta-autopilot-private-cluster 
Working in ./modules/beta-autopilot-public-cluster ...
1 issue(s) found:

Warning: [Fixable] variable "enable_default_node_pools_metadata" is declared but not used (terraform_unused_declarations)

  on variables.tf line 397:
 397: variable "enable_default_node_pools_metadata" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.7.0/docs/rules/terraform_unused_declarations.md

tflint failed ./modules/beta-autopilot-public-cluster 
```

**Comment by caueasantos** [comment]
> Thanks for the contribution @caueasantos!
> 
> Should this change apply to Autopilot cluster?
> 
> ```
> Working in ./modules/beta-autopilot-private-cluster ...
> 1 issue(s) found:
> 
> Warning: [Fixable] variable "enable_default_node_pools_metadata" is declared but not used (terraform_unused_declarations)
> 
>   on variables.tf line 427:
>  427: variable "enable_default_node_pools_metadata" {
> 
> Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.7.0/docs/rules/terraform_unused_declarations.md
> 
> tflint failed ./modules/beta-autopilot-private-cluster 
> Working in ./modules/beta-autopilot-public-cluster ...
> 1 issue(s) found:
> 
> Warning: [Fixable] variable "enable_default_node_pools_metadata" is declared but not used (terraform_unused_declarations)
> 
>   on variables.tf line 397:
>  397: variable "enable_default_node_pools_metadata" {
> 
> Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.7.0/docs/rules/terraform_unused_declarations.md
> 
> tflint failed ./modules/beta-autopilot-public-cluster 
> ```

Definitely not, let me see how I can disable the variable for autopilot clusters using the autogeneration templates.

File: autogen/main/variables.tf.tmpl [hunk 1]
```diff
@@ -619,6 +619,14 @@ variable "disable_default_snat" {
   default     = false
 }
 
+{% if autopilot_cluster != true %}
+variable "enable_default_node_pools_metadata" {
+  type        = bool
+  description = "Whether to enable the default node pools metadata key-value pairs such as `cluster_name` and `node_pool`"
+  default     = true
+}
```
Comment (apeabody) [inline autogen/main/variables.tf.tmpl:627]: nit: let's move the line return into the block so we don't get extra spaces in autopilot_cluster != true

```suggestion
}

```

File: autogen/main/variables.tf.tmpl [hunk 2]
```diff
@@ -619,6 +619,14 @@ variable "disable_default_snat" {
...
+{% endif %}
+
```
Comment (apeabody) [inline autogen/main/variables.tf.tmpl:629]: nit: let's move the line return into the block so we don't get extra spaces in autopilot_cluster != true

```suggestion
```

**Review by apeabody — APPROVED** [review]

**Pull request merged**